examples/pruning/tfkeras_integration.py

```diff
@@ -0,0 +1,144 @@
+
+    return ds
+
+
+def eval_dataset():
+
+    ds = tfds.load('mnist', split=tfds.Split.TEST, shuffle_files=False)
+    ds = ds.map(lambda x: (tf.cast(x['image'], tf.float32)/255., x['label']))
+    ds = ds.repeat().batch(BATCHSIZE)
+    ds = ds.prefetch(tf.data.experimental.AUTOTUNE)
+
+    return ds
+
+
+def create_model(trial):
+
+    # Hyperparameters to be tuned by Optuna.
+    lr = trial.suggest_loguniform('lr', 1e-4, 1e-1)
+    momentum = trial.suggest_uniform('momentum', 0.0, 1.0)
+    units = trial.suggest_categorical('units', [32, 64, 128, 256, 512])
+
+    # Compose neural network with one hidden layer.
+    model = tf.keras.Sequential()
+    model.add(tf.keras.layers.Flatten())
+    model.add(tf.keras.layers.Dense(units=units, activation=tf.nn.relu))
+    model.add(tf.keras.layers.Dense(CLASSES, activation=tf.nn.softmax))
+
+    # Compile model.
+    model.compile(
+        optimizer=tf.keras.optimizers.SGD(lr=lr, momentum=momentum, nesterov=True),
+        loss='sparse_categorical_crossentropy',
+        metrics=['accuracy'],
+    )
+
+    return model
+
+
+def objective(trial):
+
+    # Metrics to be monitored by Optuna.
+    if tf.__version__ >= '2':
+        monitor = 'val_accuracy'
+    else:
+        monitor = 'val_acc'
+
+    # Create tf.keras model instance.
+    model = create_model(trial)
+
+    # Create dataset instance.
+    ds_train = train_dataset()
+    ds_eval = eval_dataset()
+
+    # Create callbacks for early stopping and pruning.
+    callbacks = [
+        tf.keras.callbacks.EarlyStopping(patience=3),
+        TFKerasPruningCallback(trial, monitor)
+    ]
+
+    # Train model.
+    history = model.fit(
+        ds_train,
+        epochs=EPOCHS,
+        steps_per_epoch=STEPS_PER_EPOCH,
+        validation_data=ds_eval,
+        validation_steps=VALIDATION_STEPS,
+        callbacks=callbacks,
+    )
+
+    # TODO(@sfujiwara): Investigate why the logger here is called twice.
+    # tf.compat.v1.logging.set_verbosity(tf.compat.v1.logging.DEBUG)
+    # tf.compat.v1.logging.info('hello optuna')
+
+    return history.history[monitor][-1]
+
+
+def show_result(study):
+
+    pruned_trials = [t for t in study.trials if t.state == optuna.structs.TrialState.PRUNED]
+    complete_trials = [t for t in study.trials if t.state == optuna.structs.TrialState.COMPLETE]
+
+    print('Study statistics: ')
+    print('  Number of finished trials: ', len(study.trials))
+    print('  Number of pruned trials: ', len(pruned_trials))
+    print('  Number of complete trials: ', len(complete_trials))
+
+    print('Best trial:')
+    trial = study.best_trial
+
+    print('  Value: ', trial.value)
+
+    print('  Params: ')
+    for key, value in trial.params.items():
+        print('    {}: {}'.format(key, value))
+
+
+def main():
+
+    study = optuna.create_study(
+        direction='maximize',
+        pruner=optuna.pruners.MedianPruner(n_startup_trials=2)
+    )
+
+    study.optimize(objective, n_trials=10)
```

Comment:
I think that `n_trials=10` is too few (the default `TPESampler` samples parameters at random when the number of finished trials is less than 10).
How about changing the value of `n_trials` to `25`?
Besides, recently, I prefer to specify `timeout` argument to ensure that an example finishes within a reasonable time even if it is executed on a poor CPU/GPU machine.
```suggestion
    study.optimize(objective, n_trials=25, timeout=600
```